tests/framework/extensions/workloads/pods/pod_status.go

```diff
@@ -17,86 +16,80 @@ const (
-		podResults = []string{}
+		// emptying pod errors every time we poll so that we don't return stale errors
 		podErrors = []error{}
 
 		pods, err := steveClient.List(nil)
 		if err != nil {
-			podErrors = append(podErrors, err)
+			// not returning the error in this case, as it could cause a false positive if we start polling too early.
 			return false, nil
 		}
 
-		podResults = append(podResults, "Pod's Status: \n")
-
 		for _, pod := range pods.Data {
-			podResult, podError, err := CheckPodStatus(&pod)
-			if err != nil {
+			isReady, err := IsPodReady(&pod)
+			if !isReady {
 				// not returning the error in this case, as it could cause a false positive if we start polling too early.
 				return false, nil
 			}
-			if podError != nil {
-				podErrors = append(podErrors, podError)
-				return false, nil
-			}
-			if podResult != "" {
-				podResults = append(podResults, podResult)
+
+			if err != nil {
+				podErrors = append(podErrors, err)
 			}
 		}
-
-		if len(podResults) > 0 && len(podErrors) == 0 {
-			return true, nil
+		if len(podErrors) != 0 {
+			// this basically waits for all pods to be ready before returning the error
+			return false, nil
 		}
-
-		return false, nil
+		return true, nil
 	})
 
 	if err != nil {
-		return nil, []error{err}
+		return []error{err}
 	}
 
-	return podResults, podErrors
+	return podErrors
 }
 
-// CheckPodStatus is a helper function that uses the steve client to check the status of a single pod
-func CheckPodStatus(pod *v1.SteveAPIObject) (podResults string, podError error, err error) {
+func IsPodReady(pod *v1.SteveAPIObject) (bool, error) {
 	podStatus := &corev1.PodStatus{}
-	err = v1.ConvertToK8sType(pod.Status, podStatus)
+	err := v1.ConvertToK8sType(pod.Status, podStatus)
 	if err != nil {
-		return "", nil, err
+		return false, err
 	}
+
 	if podStatus.ContainerStatuses == nil || len(podStatus.ContainerStatuses) == 0 {
-		return fmt.Sprintf("WARN: %s: Container Status is Empty \n", pod.Name), nil, nil
+		return false, nil
 	}
 
-	image := podStatus.ContainerStatuses[0].Image
 	phase := podStatus.Phase
 
+	if phase == corev1.PodPending {
+		return false, nil
+	}
 	if phase == corev1.PodFailed || phase == corev1.PodUnknown {
-		logrus.Infof("Pod %s: Not active | Image %s", pod.Name, image)
-		// Do not report as error if pod is in failed state due to container termination
+		// Rancher deploys multiple hlem-operation jobs to do the same task. If one job succeeds, the others terminate with no error.
+		var errorMessage string
 		for _, containerStatus := range podStatus.ContainerStatuses {
 			if containerStatus.State.Terminated == nil {
-				return "", fmt.Errorf("ERROR: %s: %s", pod.Name, podStatus), nil
+				errorMessage += fmt.Sprintf("ERROR: %s: %s\n", pod.Name, podStatus)
 			}
 		}
-		return fmt.Sprintf("INFO: TERMINATED %s: %s\n", pod.Name, podStatus), nil, nil
-	}
-
-	if phase == corev1.PodRunning {
-		logrus.Infof("Pod %s: Active | Image: %s", pod.Name, image)
```

Comment:
Why are we removing the logging now?

Reply:
Just out of curiosity, has this been resolved?

Reply:
thought I replied here, but I removed this but can add it back if we need it. It seems a little spammy / not really adding much value to the logs in my opinion.